didcomm/did_doc/did_resolver.py

```diff
@@ -0,0 +1,48 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from didcomm.common.types import DID_OR_DID_URL
+from didcomm.did_doc.did_doc import DIDDoc
+
+
+class DIDResolver(ABC):
+    """DID Resolver to resolver a DID to a DID DOC."""
+
+    @abstractmethod
+    async def resolve(self, did: DID_OR_DID_URL) -> Optional[DIDDoc]:
```

Comment:
Can it rise exception?

Reply:
I would say no, let's keep the same signature as in the base class. So, None is returned if DID is not resolved.
It will be up to application then to raise an exception if needed. We can change it during implementation if needed.